client/src/telethon/_impl/mtsender/sender.py

```diff
@@ -189,30 +190,50 @@ async def connect(
         ip, port = addr.split(":")
         reader, writer = await connector(ip, int(port))
 
-        return cls(
+        sender = cls(
             dc_id=dc_id,
             addr=addr,
-            lock=Lock(),
             _logger=base_logger.getChild("mtsender"),
             _reader=reader,
             _writer=writer,
             _transport=transport,
             _mtp=mtp,
             _mtp_buffer=bytearray(),
+            _updates=[],
             _requests=[],
             _request_event=Event(),
-            _next_ping=asyncio.get_running_loop().time() + PING_DELAY,
             _read_buffer=bytearray(),
-            _write_drain_pending=False,
+            _step_event=Event(),
         )
 
+        sender._recv_task = asyncio.create_task(sender._do_recv())
+        sender._send_task = asyncio.create_task(sender._do_send())
+        return sender
+
     async def disconnect(self) -> None:
+        assert self._recv_task and self._send_task
+
+        (
+            recv_task,
+            send_task,
+            self._recv_task,
+            self._send_task,
+        ) = (
+            self._recv_task,
+            self._send_task,
+            None,
+            None,
+        )
```

Comment:
This looks like a very awkward way to do it.

I would simply prefer two statements:

```suggestion
        recv_task, self._recv_task = self._recv_task, None
        send_task, self._send_task = self._send_task, None
```

or:

```suggestion
        recv_task, send_task = self._recv_task, self._send_task
        self._recv_task, self._send_task = None, None
```

but the parenthesis and extra indentation looks really odd. I am more inclined towards the first. (Putting aside the fact we *still* have background tasks.)